ethers-signers/src/ledger/mod.rs

```diff
@@ -25,6 +29,16 @@ impl Signer for LedgerEthereum {
         self.sign_tx(message).await
     }
 
+    async fn sign_typed_data<T: Eip712 + Send + Sync>(
+        &self,
+        payload: T,
+        domain: Option<EIP712Domain>,
+    ) -> Result<Signature, Self::Error> {
+        let hash = payload.encode_eip712(domain)?;
```

Comment:
As I said earlier out of my comfort zone for now hehe.
But it seems that the Eip712 Error cannot be cast to a `LedgerError` here, Eventually we can handle the Result differently?
```
the trait `From<<T as Eip712>::Error>` is not implemented for `ledger::types::LedgerError`
```